fiberplane-models/src/webhooks.rs

```diff
@@ -0,0 +1,188 @@
+// because only 0 and 1 are covered cases and std would have no idea how to handle that
+#[allow(clippy::from_over_into)]
+impl TryInto<WebhookCategory> for i16 {
+    type Error = InvalidWebhookIdError;
+
+    fn try_into(self) -> Result<WebhookCategory, Self::Error> {
+        match self {
+            0 => Ok(WebhookCategory::Ping),
+            1 => Ok(WebhookCategory::FrontMatter),
+            value => Err(InvalidWebhookIdError(value)),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "fp-bindgen",
+    derive(Serializable),
+    fp(rust_module = "fiberplane_models::webhooks")
+)]
+#[error("unknown value {0}, expected 0 (ping) or 1 (front_matter)")]
+pub struct InvalidWebhookIdError(i16);
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize, TypedBuilder)]
+#[cfg_attr(
+    feature = "fp-bindgen",
+    derive(Serializable),
+    fp(rust_module = "fiberplane_models::webhooks")
+)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Webhook {
+    #[builder(setter(into))]
+    pub id: Base64Uuid,
+    #[builder(setter(into))]
+    pub workspace_id: Base64Uuid,
+    #[builder(setter(into))]
+    pub endpoint: String,
+    pub events: Vec<WebhookCategory>,
+    #[builder(default, setter(strip_option, into))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub shared_secret: Option<String>,
+    #[builder(default)]
+    pub enabled: bool,
+    #[builder(default, setter(strip_option, into))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub created_by: Option<Base64Uuid>,
+    #[builder(setter(into))]
+    pub created_at: Timestamp,
+    #[builder(setter(into))]
+    pub updated_at: Timestamp,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize, TypedBuilder)]
+#[cfg_attr(
+    feature = "fp-bindgen",
+    derive(Serializable),
+    fp(rust_module = "fiberplane_models::webhooks")
+)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct NewWebhook {
+    #[builder(setter(into))]
+    pub endpoint: String,
+    pub events: Vec<WebhookCategory>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize, TypedBuilder)]
+#[cfg_attr(
+    feature = "fp-bindgen",
+    derive(Serializable),
+    fp(rust_module = "fiberplane_models::webhooks")
+)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct UpdateWebhook {
+    #[builder(default, setter(strip_option, into))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub endpoint: Option<String>,
+    #[builder(default, setter(strip_option))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub events: Option<Vec<WebhookCategory>>,
+    #[builder(default)]
+    #[serde(default)]
+    pub regenerate_shared_secret: bool,
+    #[builder(default, setter(strip_option))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub enabled: Option<bool>,
```

Comment:
Maybe we can add some newlines in between the properties to make it a bit more readable.

```suggestion
    #[builder(default, setter(strip_option, into))]
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub endpoint: Option<String>,

    #[builder(default, setter(strip_option))]
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub events: Option<Vec<WebhookCategory>>,

    #[builder(default)]
    #[serde(default)]
    pub regenerate_shared_secret: bool,

    #[builder(default, setter(strip_option))]
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub enabled: Option<bool>,
```

Also applies to the other structs